### PR TITLE
feat: add total consumption

### PR DESCRIPTION
**Pull request by johevemi**
## References

* #1125 #1054

### CODE REVIEW

File: source/databricks/calculation-engine/package/balance_fixing.py [hunk 1]
```diff
@@ -105,11 +105,14 @@ def _calculate(
         grid_loss_responsible_df
     )
 
-    _calculate_non_profiled_consumption(
+    consumption_per_ga = _calculate_non_profiled_consumption(
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/balance_fixing.py:108]: non_profiled_consumption_per_ga?

File: source/databricks/calculation-engine/package/balance_fixing.py [hunk 2]
```diff
@@ -105,11 +105,14 @@ def _calculate(
...
         actors_writer, result_writer, consumption_per_ga_and_brp_and_es
     )
-    _calculate_production(actors_writer, result_writer, production_per_ga_and_brp_and_es)
-    _calculate_flex_consumption(actors_writer, result_writer, flex_consumption_per_ga_and_brp_and_es)
+    production_per_ga = _calculate_production(actors_writer, result_writer, production_per_ga_and_brp_and_es)
+    flex_consumption_per_ga = _calculate_flex_consumption(actors_writer, result_writer, flex_consumption_per_ga_and_brp_and_es)
+
+    _calculate_total_consumption(actors_writer, result_writer, production_per_ga, net_exchange_per_ga)
+    _calculate_residual_ga(net_exchange_per_ga, consumption_per_ga, flex_consumption_per_ga, production_per_ga)
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/balance_fixing.py:115]: Why is this needed?

File: source/databricks/calculation-engine/package/balance_fixing.py (hunk 1 above)
Comment (johevemi) [reply source/databricks/calculation-engine/package/balance_fixing.py:108]: removed

File: source/databricks/calculation-engine/package/balance_fixing.py (hunk 2 above)
Comment (johevemi) [reply source/databricks/calculation-engine/package/balance_fixing.py:115]: removed

File: source/databricks/calculation-engine/package/steps/aggregation/grid_loss_calculator.py [hunk 3]
```diff
@@ -46,11 +46,12 @@ def calculate_grid_loss(
 
 
 # (step 22)
-def calculate_residual_ga(results: dict) -> DataFrame:
-    agg_net_exchange = results[ResultKeyName.net_exchange_per_ga]
-    agg_non_profiled_consumption = results[ResultKeyName.non_profiled_consumption_ga]
-    agg_flex_consumption = results[ResultKeyName.flex_consumption_ga]
-    agg_production = results[ResultKeyName.production_ga]
+def calculate_residual_ga(
```
Comment (JesperGomes) [inline source/databricks/calculation-engine/package/steps/aggregation/grid_loss_calculator.py:49]: is this needed?

Comment (johevemi) [reply source/databricks/calculation-engine/package/steps/aggregation/grid_loss_calculator.py:49]: Removed

Comment (JesperGomes) [reply source/databricks/calculation-engine/package/steps/aggregation/grid_loss_calculator.py:49]: is it?

Comment (johevemi) [reply source/databricks/calculation-engine/package/steps/aggregation/grid_loss_calculator.py]: ah no, now it is

Comment (johevemi) [reply source/databricks/calculation-engine/package/steps/aggregation/grid_loss_calculator.py]: also the test for it

**Review by JesperGomes — APPROVED** [review]
